src/odemis/driver/lakeshore.py

```diff
@@ -344,98 +405,228 @@ def HeaterSetup(self,
+
+        self._serial = self._openSerialPort(port)
+
+    def terminate(self):
+        with self._ser_access:
+            self._serial.close()
+
+    def sendOrder(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF)
+        """
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+
+    def sendQuery(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF, but with the ?)
+        returns (byte str): answer received from the device (without \n or \r)
+        raise:
+            IOError if no answer is returned in time
+        """
+
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._serial.write(cmd)
+
+            self._serial.timeout = 1
+            ans = b''
+            while ans[-1:] != b'\n':
+                char = self._serial.read()
+                if not char:
+                    raise IOError("Timeout after receiving %s" % to_str_escape(ans))
+                ans += char
+
+            logging.debug("Received answer %s", to_str_escape(ans))
+
+            return ans.strip()
+
+
+class IPBusAccesser(object):
+    """
+    Manage TCP/IP connections over ethernet
+    """
+
+    def __init__(self, ip_address: str):
+        """
+        ip_address (str): IP address to establish the ethernet connection
+        """
+        self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._socket.settimeout(3)  # set 3 seconds timeout to receive data
+        try:
+            self._socket.connect((ip_address, MODEL_350_TCP_PORT))
+        except socket.timeout:
+            raise model.HwError("Connection is timed out.")
+
+        self._ser_access = threading.Lock()
+
+    def terminate(self):
+        self._socket.close()
+
+    def sendOrder(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF)
+        """
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._socket.send(cmd)
+
+    def sendQuery(self, cmd):
+        """
+        cmd (byte str): command to be sent to device (without the LF, but with the ?)
+        returns (byte str): answer received from the device (without \n or \r)
+        raise:
+            IOError if no answer is returned in time
+        """
+
+        cmd = cmd + b"\n"
+        with self._ser_access:
+            logging.debug("Sending command %s", to_str_escape(cmd))
+            self._socket.send(cmd)
+
+            ans = b''
+            while ans[-1:] != b'\n':
+                try:
+                    data = self._socket.recv(TCP_BUFFER_SIZE)
+                except socket.timeout:
+                    raise model.HwError("Connection is timed out.")
```

Comment:
Here, I'd suggest to just raise a IOError. HwError is just for certain class of recoverable errors.